cli/azd/pkg/ux/action_printer.go

```diff
@@ -0,0 +1,96 @@
+package ux
+
+import (
+	"context"
+)
+
+// Creates a new instance of an ActionPrinter
+func NewActionPrinter[R any]() *ActionPrinter[R] {
```

Comment:
I am wondering if we can recover good typing my making this have another generic argument, which is the type of the StepContext parameter and then thread that type parameter down to the Step interface, which itself becomes generic.

Reply:
Thanks for the feedback.  I went back and forth on generics a bit.  We'll need to share data between steps and wasn't sure a single generic value would be good enough and opted for a map like struct wrapped in a context like object. 

Step 1 may do something that step 2 needs.  Step 3 needs stuff from step 2 but doesn't really care about what happened in step 1.

Will need to output data in the complete step for success and error scenarios.

Reply:
> Thanks for the feedback. I went back and forth on generics a bit. We'll need to share data between steps and wasn't sure a single generic value would be good enough and opted for a map like struct wrapped in a context like object.

It seems like a generic value is more powerful than the maplike object, since I could always instantiate the generic with `StepContext` if a I wanted, or instead something like 

```golang
struct {
    Step1Data string
    Step2Data int
}
```

> Will need to output data in the complete step for success and error scenarios.

Guessing there might be something clever we can do here with interface that defines a method that gives the information Complete needs, and then use constraints to ensure the context object has an implementation of that interface.

I'm just trying to avoid a bunch of type assertions to recover go types from data between steps - but honestly I'm not sure how much context data we'll end up passing between steps, so in practice maybe this won't be so bad.

Happy to go with what you have here, but if the step implementations start to look gnarly maybe we should think about generics again.